File: backend/app/children/domain/school_class.py
from uuid import UUID

from pydantic.dataclasses import dataclass

from app.children.domain.school import School
from app.shared.domain import Entity, ValueObject


@dataclass(eq=True, frozen=True)
class ClassLiteral(ValueObject):
    value: str


@dataclass(eq=True, frozen=True)
class ClassNumber(ValueObject):
    value: int


@dataclass
class SchoolClass(Entity):
    id: UUID
    school: School
    number: ClassNumber
    literal: ClassLiteral

File: frontend/web/src/7_shared/api/implementations/v3/frontend-types/nutrition/school-class.ts
import {Mealtime} from './mealtime.ts';

export type GetSchoolClassesParams = {
  teacherId: string
}

export type SchoolClass = {
  id: string,
  teacherId: string,
  number: number,
  literal: string,
  mealtimes: Mealtime[]
}
